Simplified/AxisService/NYPLAxisLicenseService.swift

```diff
@@ -0,0 +1,190 @@
+/// license is valid.
+/// 
+struct NYPLAxisLicenseService: NYPLAxisLicenseHandling {
+  
+  static private let licenseValidationFailureSummary = "Axis failed to validate license"
+  
+  let axisItemDownloader: NYPLAxisItemDownloading
+  let axisKeysProvider: NYPLAxisKeysProviding
+  let bookVaultId: String
+  let cypher: NYPLRSACryptographing
+  let deviceInfoProvider: NYPLDeviceInfoProviding
+  let isbn: String
+  let parentDirectory: URL
+  let localLicenseURL: URL
+  
+  init(axisItemDownloader: NYPLAxisItemDownloading,
+       axisKeysProvider: NYPLAxisKeysProviding = NYPLAxisKeysProvider(),
+       bookVaultId: String,
+       cypher: NYPLRSACryptographing,
+       deviceInfoProvider: NYPLDeviceInfoProviding = NYPLAxisDRMAuthorizer.sharedInstance,
+       isbn: String,
+       parentDirectory: URL) {
+    
+    self.axisItemDownloader = axisItemDownloader
+    self.axisKeysProvider = axisKeysProvider
+    self.bookVaultId = bookVaultId
+    self.cypher = cypher
+    self.deviceInfoProvider = deviceInfoProvider
+    self.isbn = isbn
+    self.parentDirectory = parentDirectory
+    self.localLicenseURL = parentDirectory
+      .appendingPathComponent(axisKeysProvider.desiredNameForLicenseFile)
+  }
+  
+  func downloadLicense() {
+    axisItemDownloader.dispatchGroup.enter()
+    
+    let licenseURL = NYPLAxisLicenseURLGenerator(
+      baseURL: axisKeysProvider.licenseBaseURL,
+      bookVaultId: bookVaultId,
+      clientIP: deviceInfoProvider.clientIP,
+      cypher: cypher,
+      deviceID: deviceInfoProvider.deviceID,
+      isbn: isbn
+    ).generateLicenseURL()
+    
+    axisItemDownloader.downloadItem(from: licenseURL, at: localLicenseURL)
+  }
+  
+  /// Validates license by decrypting the value contained in `[encryption][user_key][key_check]`
+  /// using our private key and compares it against the bookVaultId we sent while requesting the license.
+  ///
+  /// - Note: We delete the license file upon validation in order to adhere to Axis guidelines. We do
+  /// however store the book_vault_id and isbn for downloading license again when the user wants to open
+  /// the book for reading.
+  func validateLicense() {
+    axisItemDownloader.dispatchGroup.wait()
+    axisItemDownloader.dispatchGroup.enter()
+    guard
+      let data = try? Data(contentsOf: localLicenseURL),
+      let jsonObject = try? JSONSerialization.jsonObject(with: data, options: .fragmentsAllowed),
+      let json = jsonObject as? [String: Any],
+      let encryptionNode = json["encryption"] as? [String: Any],
+      let userKey = encryptionNode["user_key"] as? [String: Any],
+      let encryptedBookVaultId = userKey["key_check"] as? String,
+      let base64Data = Data(base64Encoded: encryptedBookVaultId),
+      let decryptedBookVaultData = cypher.decryptWithPKCS1_OAEP(base64Data),
+      let decryptedBookVaultId = String(data: decryptedBookVaultData, encoding: .utf8)
+    else {
+      logLicenseError(reason: "Axis failed to validate license")
```

Comment:
The string is already defined as `licenseValidationFailureSummary`, so we should use that instead.
Same for below.

Reply:
Great catch. Done